sdk/src/main/java/com/horizen/account/transaction/EthereumTransaction.java

```diff
@@ -0,0 +1,412 @@
+        if (getValue().signum() < 0)
+            throw new TransactionSemanticValidityException(String.format("Transaction [%s] is semantically invalid: " +
+                    "negative value", id()));
+        if (getNonce().signum() < 0)
+            throw new TransactionSemanticValidityException(String.format("Transaction [%s] is semantically invalid: " +
+                    "negative nonce", id()));
+        if (getGasLimit().signum() <= 0)
+            throw new TransactionSemanticValidityException(String.format("Transaction [%s] is semantically invalid: " +
+                    "non-positive gas limit", id()));
+        if (!BigIntegerUtil.isUint64(getGasLimit())) throw new GasUintOverflowException();
+        //TODO why this check is not in Geth (maybe)?
+        if (getTo() == null && getData().length == 0)
+            throw new TransactionSemanticValidityException(String.format("Transaction [%s] is semantically invalid: " +
+                    "smart contract declaration transaction without data", id()));
+
+        if (getMaxFeePerGas().signum() < 0)
+            throw new TransactionSemanticValidityException(String.format("Transaction [%s] is semantically invalid: " +
+                    "eip1559 transaction with negative maxFeePerGas", id()));
+        if (getMaxPriorityFeePerGas().signum() < 0)
+            throw new TransactionSemanticValidityException(String.format("Transaction [%s] is semantically invalid: " +
+                    "eip1559 transaction with negative maxPriorityFeePerGas", id()));
+        if (getMaxFeePerGas().bitLength() > 256)
+            throw new TransactionSemanticValidityException(String.format("Transaction [%s] is semantically invalid: " +
+                    "eip1559 transaction maxFeePerGas bit length [%d] is too high", id(), getMaxFeePerGas().bitLength()));
+
+        if (getMaxPriorityFeePerGas().bitLength() > 256)
+            throw new TransactionSemanticValidityException(String.format("Transaction [%s] is semantically invalid: " +
+                    "eip1559 transaction maxPriorityFeePerGas bit length [%d] is too high", id(), getMaxPriorityFeePerGas().bitLength()));
+
+        if (getMaxFeePerGas().compareTo(getMaxPriorityFeePerGas()) < 0)
+            throw new TransactionSemanticValidityException(String.format("Transaction [%s] is semantically invalid: " +
+                            "eip1559 transaction max priority fee per gas [%s] higher than max fee per gas [%s]",
+                    id(), getMaxPriorityFeePerGas(), getMaxFeePerGas()));
+
+        if (getGasLimit().compareTo(GasUtil.intrinsicGas(getData(), getTo() == null)) < 0) {
+            throw new TransactionSemanticValidityException(String.format("Transaction [%s] is semantically invalid: " +
+                    "gas limit is below intrinsic gas", id()));
+        }
+
+        try {
+            if (this.getFrom() == null || this.getFrom().address().length != Account.ADDRESS_SIZE)
```

Comment:
We detect unsigned transactions here, if `getFrom() == null`. In this case however the error message would  not be clear, I propose to check directly `isSigned() `and possibly doing this check at the very beginning of the method.